clients/java/src/main/java/io/camunda/client/api/search/filter/ProcessInstanceFilter.java

```diff
@@ -122,4 +124,19 @@ ProcessInstanceFilter variables(
 
   /** Filter by error message using {@link StringProperty} consumer */
   ProcessInstanceFilter errorMessage(final Consumer<StringProperty> fn);
+
+  /** Filter by flowNodeId */
+  ProcessInstanceFilter flowNodeId(final String flowNodeId);
+
+  /** Filter by flowNodeId using {@link StringProperty} */
+  ProcessInstanceFilter flowNodeId(final Consumer<StringProperty> fn);
+
+  /** Filter by state */
+  ProcessInstanceFilter flowNodeInstanceState(final FlowNodeInstanceState state);
+
+  /** Filter by flowNodeInstanceState using {@link FlowNodeInstanceStateProperty} */
+  ProcessInstanceFilter flowNodeInstanceState(final Consumer<FlowNodeInstanceStateProperty> fn);
+
+  /** Filter by hasFlowNodeInstanceIncident */
+  ProcessInstanceFilter hasFlowNodeInstanceIncident(final Boolean hasFlowNodeInstanceIncident);
```

Comment:
💭 Maybe just prefix them with `flowNode` instead of `flowNodeInstance`? It would fit better with `flowNodeId`.

- `flowNodeState`
- `hasFlowNodeIncident`

What do you think?